internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -251,41 +261,117 @@ func convertEFSPermissions(input map[string]manifest.Volume) ([]*template.EFSPer
 		if volume.EFS == nil {
 			continue
 		}
+		if volume.EFS.UseManagedFS() {
+			continue
+		}
 		// Write defaults to false.
 		write := defaultWritePermission
 		if volume.ReadOnly != nil {
 			write = !aws.BoolValue(volume.ReadOnly)
 		}
-		if volume.EFS.FileSystemID == nil {
+
+		id, err := volume.EFS.FSID()
+		if err != nil {
+			return nil, err
+		}
+		if id == nil {
 			return nil, errNoFSID
 		}
+
 		var accessPointID *string
-		if volume.EFS.AuthConfig != nil {
-			accessPointID = volume.EFS.AuthConfig.AccessPointID
+		if volume.EFS.Config.AuthConfig != nil {
+			accessPointID = volume.EFS.Config.AuthConfig.AccessPointID
 		}
 		perm := template.EFSPermission{
 			Write:         write,
 			AccessPointID: accessPointID,
-			FilesystemID:  volume.EFS.FileSystemID,
+			FilesystemID:  id,
 		}
 		output = append(output, &perm)
 	}
 	return output, nil
 }
 
-func convertVolumes(input map[string]manifest.Volume) ([]*template.Volume, error) {
-	if len(input) == 0 {
-		return nil, nil
+func convertManagedFSInfo(wl manifest.Workload, input map[string]manifest.Volume) (*template.ManagedVolumeCreationInfo, error) {
+	var output *template.ManagedVolumeCreationInfo
+	for name, volume := range input {
+		if output != nil {
+			return nil, fmt.Errorf("validate managed EFS: cannot specify more than one managed volume per service")
+		}
+		if volume.EFS == nil {
+			continue
+		}
+
+		if volume.EFS.UseManagedFS() {
+			uid := volume.EFS.Config.UID
+			gid := volume.EFS.Config.GID
+			dirName := getDirName(wl)
+
+			if err := validateUIDGID(uid, gid); err != nil {
+				return nil, err
+			}
+
+			if uid == nil && gid == nil {
+				crc := aws.Uint32(getRandomUIDGID(dirName))
+				uid = crc
+				gid = crc
+			}
+			output = &template.ManagedVolumeCreationInfo{
+				Name:    aws.String(name),
+				DirName: aws.String(dirName),
+				UID:     uid,
+				GID:     gid,
+			}
+		}
+	}
+	return output, nil
+}
+
+// getRandomUIDGID returns the 32-bit checksum of the service name for use as CreationInfo in the EFS Access Point.
+// See https://stackoverflow.com/a/14210379/5890422 for discussion of the possibility of collisions in CRC32 with
+// small numbers of hashes.
+func getRandomUIDGID(name string) uint32 {
+	return crc32.ChecksumIEEE([]byte(name))
+}
+
+func getDirName(wl manifest.Workload) string {
+	for _, w := range manifest.ServiceTypes {
+		if aws.StringValue(wl.Type) == w {
+			return svcWlType + "/" + aws.StringValue(wl.Name)
```

Comment:
Actually do we need the `workloadType/` prefix path? it looks like we store all these workloads under `components/` in SSM so it shouldn't be possible to declare a job with the same name as a service

Reply:
Huh, this is exposing an odd issue where we don't actually error if the name is taken. For example, in a workspace where I already have a LB Web Service called `fe`, I can do the following:

```
$ copilot job init -n fe -d ./Dockerfile --schedule "@every 90m"

✔ Manifest file for job fe already exists at copilot/fe/manifest.yml, skipping writing it.
Your manifest contains configurations like your container size and job schedule (@every 90m).

✔ Created ECR repositories for job fe.

Recommended follow-up actions:
- Update your manifest copilot/fe/manifest.yml to change the defaults.
- Run `copilot job deploy --name fe --env test` to deploy your job to a test environment.

$ copi job deploy
✘ select job: no jobs found
```

I've captured this in #2156.

But it does look like we don't need the prefix after all!

Reply:
Removed the prefix.